ecl/hthor/hthor.cpp

```diff
@@ -8369,6 +8369,9 @@ bool CHThorDiskReadBaseActivity::openNext()
 #ifdef _DEBUG
                                 EXCLOG(e, nullptr);
 #endif
+                                VStringBuffer msg("Remote streaming failure, failing over to direct read for: '%s'. ", file.str());
+                                e->errorMessage(msg);
+                                agent.addWuException(msg.str(), WRN_RemoteReadFailure, SeverityWarning, "hthor");
```

Comment:
@shamser note: this is an example of a warning that it would be very useful to associate with the activity in wudetails.

Reply:
FWIW, in Thor's case for this PR, the exception text looks like:
e.g.: 
```
  <Exception activity="18"
             code="10139"
             sequence="0"
             severity="1"
             source="SLAVE #3"
             time="2019-06-17T10:15:41Z">
   Graph graph3[17], indexread[18]: SLAVE #3 [192.168.9.108:20140]: Remote streaming failure, failing over to direct read for: &apos;/home/jsmith/hpccdeb/var/lib/HPCCSystems/hpcc-data/thor/regress/single/thorlcr/w20190617-111539/indexlimits_index._3_of_5&apos;. /home/jsmith/hpccdeb/var/lib/HPCCSystems/hpcc-data/thor/regress/single/thorlcr/w20190617-111539/indexlimits_index._3_of_5[192.168.9.108:7100] ERROR: cmd=RFCStreamRead, error=Internal Error (3000, assert(length &lt;= reserved) failed - file: rtlds.cpp, line 1889)
  </Exception>
```

Eclwatch seems to currently parse the text and provide a hyperlink to the graph/activity.
But presumably HPCC-22342 would seek to explicitly set the scope (not in the text of the message).